lib/shopify_app/controller_concerns/with_shopify_id_token.rb

```diff
@@ -5,31 +5,30 @@ module WithShopifyIdToken
     extend ActiveSupport::Concern
 
     def shopify_id_token
-      @shopify_id_token ||= id_token_from_request_env || id_token_from_authorization_header || id_token_from_url_param
+      @shopify_id_token ||= id_token_from_authorization_header || id_token_from_url_param
+    end
+
+    def jwt_payload
+      @jwt_payload ||= shopify_id_token.present? ? ShopifyAPI::Auth::JwtPayload.new(shopify_id_token) : nil
```

Comment:
Is there some way we can skip parsing the JWT token twice, once here, and once in https://github.com/Shopify/shopify-api-ruby/blob/a64f7dd391306033ca58399b42fa32207ba6dc34/lib/shopify_api/utils/session_utils.rb#L48?

(That could be a follow-up optimization).

Reply:
Good idea. Looks like that would be a change in the `shopify-api-ruby` repo, so will leave it as a follow-up.